src/runtime/virtcontainers/fs_share_linux.go

```diff
@@ -502,11 +512,37 @@ func handleVirtualVolume(c *Container) ([]*grpc.Storage, string, error) {
 
 func (f *FilesystemShare) shareRootFilesystemWithVirtualVolume(ctx context.Context, c *Container) (*SharedFile, error) {
 	guestPath := filepath.Join("/run/kata-containers/", c.id, c.rootfsSuffix)
-	rootFsStorages, _, err := handleVirtualVolume(c)
+	rootFsStorages, volumeType, err := handleVirtualVolume(c)
 	if err != nil {
 		return nil, err
 	}
 
+	if volumeType == types.KataVirtualVolumeImageRawBlockType || volumeType == types.KataVirtualVolumeLayerRawBlockType {
+		kataGuestDir := filepath.Join(defaultKataGuestVirtualVolumedir, "containers")
+		rootfs := &grpc.Storage{}
+		rootfs.MountPoint = guestPath
+		overlayDirDriverOption := "io.katacontainers.volume.overlayfs.create_directory"
+		rootfs.Source = typeOverlayFS
```

Comment:
This line looks wrong, based on the line below it...?

Reply:
Thanks. Please could you push your latset branch as it's still showing the wrong value.